pkgs/README.md

```diff
@@ -459,7 +459,9 @@ Examples going from bad to best practices:
 
 ## Patches
 
-Patches available online should be retrieved using `fetchpatch`.
+Sometimes, changes are needed to the source to allow building a derivation in nixpkgs, or to get earlier access to an upstream fix or improvement. When using the `patches` parameter to `mkDerivation`, make sure the patch name clearly describes the reason for the patch, or add a comment.
+
+Patches already merged upstream or published elsewhere should be retrieved using `fetchpatch`.
```

Comment:
```suggestion
Patches already merged upstream, reviewed or published elsewhere should be retrieved using `fetchpatch`.
```

Reply:
Thanks for your suggestions! I have merged the others but not this one, as I don't think it makes sense to `fetchpatch` patches that have been reviewed but not merged: they might still get rebased/squashed and thus disappear..